stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionResult.kt

```diff
@@ -18,8 +19,17 @@ internal sealed class PaymentOptionResult(
 
     @Parcelize
     data class Succeeded(
-        val paymentSelection: PaymentSelection
-    ) : PaymentOptionResult(Activity.RESULT_OK)
+        val paymentSelection: PaymentSelection,
+        val newSavedPaymentMethod: PaymentMethod? = null,
+    ) : PaymentOptionResult(Activity.RESULT_OK) {
+        init {
+            val isSavedNewCard =
+                (paymentSelection as? PaymentSelection.New.Card)?.shouldSavePaymentMethod == true
+            if (isSavedNewCard) {
+                requireNotNull(newSavedPaymentMethod)
+            }
```

Comment:
It feels like we should create a new `PaymentOptionResult` sealed type if we want to enforce that `newSavedPaymentMethod` should be not-null.